### PR TITLE
support plain A record if ingress is using IP

### PR DESCRIPTION
**Pull request by ideahitme**
/cc @linki @hjacobs @tuxlife

### CODE REVIEW

File: consumers/aws.go [hunk 1]
```diff
@@ -300,11 +300,11 @@ func (a *awsConsumer) getRecordTarget(r *route53.ResourceRecordSet) string {
 	return aws.StringValue(r.ResourceRecords[0].Value)
 }
 
-//endpointsToAlias converts pkg Endpoint to route53 Alias Records
-func (a *awsConsumer) endpointsToAlias(endpoints []*pkg.Endpoint) ([]*route53.ResourceRecordSet, error) {
-	lbDNS := make([]string, len(endpoints))
+//endpointsToRecords converts pkg Endpoint to route53 Alias Records
```
Comment (ankon) [inline consumers/aws.go]: "Alias" here in the description should then also go?

Comment (ideahitme) [reply consumers/aws.go]: @ankon yes this WIP ;)

File: consumers/aws.go [hunk 2]
```diff
@@ -314,24 +314,35 @@ func (a *awsConsumer) endpointsToAlias(endpoints []*pkg.Endpoint) ([]*route53.Re
 
 	for _, ep := range endpoints {
 		if loadBalancerZoneID, exist := zoneIDs[ep.Hostname]; exist {
-			rset = append(rset, a.endpointToAlias(ep, aws.String(loadBalancerZoneID)))
+			rset = append(rset, a.endpointToRecord(ep, aws.String(loadBalancerZoneID)))
+		} else if ep.IP != "" {
+			rset = append(rset, a.endpointToRecord(ep, aws.String("")))
 		} else {
 			log.Errorf("Canonical Zone ID for endpoint: %s is not found", ep.Hostname)
```
Comment (hjacobs) [inline consumers/aws.go]: wording: "xy was not found"

File: consumers/aws.go [hunk 3]
```diff
@@ -300,11 +301,11 @@ func (a *awsConsumer) getRecordTarget(r *route53.ResourceRecordSet) string {
 	return aws.StringValue(r.ResourceRecords[0].Value)
 }
 
-//endpointsToAlias converts pkg Endpoint to route53 Alias Records
-func (a *awsConsumer) endpointsToAlias(endpoints []*pkg.Endpoint) ([]*route53.ResourceRecordSet, error) {
-	lbDNS := make([]string, len(endpoints))
+//endpointsToRecords converts pkg Endpoint to route53 A [Alias] Records depending whether IP/LB Hostname is used
+func (a *awsConsumer) endpointsToRecords(endpoints []*pkg.Endpoint) ([]*route53.ResourceRecordSet, error) {
+	lbDNS := make([]string, 0)
```
Comment (linki) [inline consumers/aws.go]: if you don't filter you can set the `capacity` here.

**Comment by linki** [comment]
LGTM

File: consumers/aws.go [hunk 4]
```diff
@@ -300,11 +301,11 @@ func (a *awsConsumer) getRecordTarget(r *route53.ResourceRecordSet) string {
...
 	for i := range endpoints {
-		lbDNS[i] = endpoints[i].Hostname
+		lbDNS = append(lbDNS, endpoints[i].Hostname)
```
Comment (tuxlife) [inline consumers/aws.go]: Why did you use the index (`i`) an not the value?
```GO
for _, endpoint := range endpoints {
    lbDNS = append(lbDNS, endpoint.Hostname)
}
```

File: consumers/aws.go [hunk 5]
```diff
@@ -103,11 +104,11 @@ func (a *awsConsumer) syncPerHostedZone(kubeRecords []*route53.ResourceRecordSet
 	upsertedMap := make(map[string]bool) // keep track of records to be upserted
 	targetMap := map[string][]*string{}  // map dnsname -> list of targets
 	for _, kr := range kubeRecords {
-		targetMap[aws.StringValue(kr.Name)] = append(targetMap[aws.StringValue(kr.Name)], kr.AliasTarget.DNSName)
+		targetMap[aws.StringValue(kr.Name)] = append(targetMap[aws.StringValue(kr.Name)], aws.String(a.getRecordTarget(kr)))
```
Comment (tuxlife) [inline consumers/aws.go:107]: Why did you use `aws.StringValue(kr.Name)` instead of `*kr.Name`

File: consumers/aws.go (hunk 4 above)
Comment (ideahitme) [reply consumers/aws.go]: right, left over from previous loop impl where index was a better option :)

File: consumers/aws.go (hunk 5 above)
Comment (ideahitme) [reply consumers/aws.go:107]: safeguard if `kr.Name` is nil

File: consumers/aws_test.go [hunk 6]
```diff
@@ -27,17 +27,40 @@ func TestEndpointToAlias(t *testing.T) {
...
 		groupID: groupID,
 	}
+	//both Hostname and IP specified -> Alias Record
 	ep := &pkg.Endpoint{
 		DNSName:  "example.com",
 		IP:       "10.202.10.123",
 		Hostname: "amazon.elb.com",
 	}
-	rsA := client.endpointToAlias(ep, &zoneID)
+	rsA := client.endpointToRecord(ep, &zoneID)
```
Comment (szuecs) [inline consumers/aws_test.go:36]: then the TestName should be fixed, too :)

**Comment by ideahitme** [comment]
if all is good, this can be merged IMO

**Comment by tuxlife** [comment]
:+1:

**Pull request merged**